File: svc_Core/Bank.Core.App/Dto/AccountDto.cs
using Bank.Core.Domain;

namespace Bank.Core.App.Dto;

public class AccountDto
{
    public Guid Id { get; set; }
    public long Balance { get; set; }
    public string Name { get; set; }

    public static AccountDto From(Account account) =>
        new AccountDto
        {
            Id = account.Id,
            Balance = account.Balance,
            Name = account.Name
        };
}

File: client_Customer/app/src/main/java/com/example/customerclient/data/api/dto/AccountDto.kt
package com.example.customerclient.data.api.dto

import com.example.customerclient.ui.home.BillInfo

data class AccountDto(
    val id: String,
    val balance: MoneyDto,
    val name: String?
)

fun AccountDto.toBillInfo() = BillInfo(
    id = this.id,
    name = this.name ?: "",
    currency = this.balance.currency,
    balance = "${this.balance.amount} ${this.balance.currency}",
    type = "Сберегательный счёт",
    duration = "бессрочный"
)
